From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Thu, 13 Aug 2026 16:24:37 +0200
Subject: ARM: dts: sun5i: pocketbook-touch-lux-3: Use the upstream tps65185
 binding

Replace the hacky single-regulator tp65185x node with the upstream
ti,tps65185 binding: split v3p3/vposneg/vcom regulators, upstream gpio
names, INT line on PB10 enabled (used for the temperature sensor
conversion irq). VCOM is settable in a 2.0-4.0V window (magnitude)
around the panel's nominal 3.12V, which the display node carries in
the new vcom-microvolt property; waveform files ship per-block vcom
offsets that need to be applied at runtime.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts | 49 +++++++---
 1 file changed, 35 insertions(+), 14 deletions(-)

diff --git a/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts b/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts
+++ b/arch/arm/boot/dts/allwinner/sun5i-a13-pocketbook-touch-lux-3.dts
@@ -130,24 +130,41 @@ pcf8563: rtc@51 {
 		#clock-cells = <0>;
 	};
 
-	// hacky PMIC driver for eInk display
-	tp65185x: regulator@68 {
-		compatible = "tp,tp65185x";
+	// eInk display PMIC
+	tps65185: pmic@68 {
+		compatible = "ti,tps65185";
 		reg = <0x68>;
 
-		regulator-min-microvolt = <3120000>;
-		regulator-max-microvolt = <3120000>;
-		regulator-name = "eink-vcom";
-
-		//XXX: we don't use interrupts in the driver right now
-		//interrupt-parent = <&pio>;
-		//interrupts = <1 10 IRQ_TYPE_LEVEL_LOW>; /* PB10 */
+		interrupt-parent = <&pio>;
+		interrupts = <1 10 IRQ_TYPE_LEVEL_LOW>; /* PB10 */
 
 		wakeup-gpios = <&pio 3 18 GPIO_ACTIVE_HIGH>; /* PD18 */
-		powerup-gpios = <&pio 4 7 GPIO_ACTIVE_HIGH>; /* PE7 */
-		vcom-gpios = <&pio 3 19 GPIO_ACTIVE_HIGH>; /* PD19 */
+		enable-gpios = <&pio 4 7 GPIO_ACTIVE_HIGH>; /* PE7 */
+		vcom-ctrl-gpios = <&pio 3 19 GPIO_ACTIVE_HIGH>; /* PD19 */
 		//XXX: not a pullup on BSP (might have an external resistor)
-		powergood-gpios = <&pio 4 6 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>; /* PE6 */
+		pwr-good-gpios = <&pio 4 6 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>; /* PE6 */
+
+		vin-supply = <&reg_vcc3v3>;
+
+		regulators {
+			tps_v3p3: v3p3 {
+				regulator-name = "eink-v3p3";
+			};
+
+			tps_vposneg: vposneg {
+				regulator-name = "eink-vposneg";
+				regulator-min-microvolt = <15000000>;
+				regulator-max-microvolt = <15000000>;
+			};
+
+			tps_vcom: vcom {
+				regulator-name = "eink-vcom";
+				/* magnitude; reasonable range around the
+				 * panel's -3.12V for waveform vcom offsets */
+				regulator-min-microvolt = <2000000>;
+				regulator-max-microvolt = <4000000>;
+			};
+		};
 	};
 };
 
@@ -308,7 +325,11 @@ display-controller@1c0c000 {
 		interconnects = <&mbus 18>;
 		interconnect-names = "dma-mem";
 
-		panel-supply = <&tp65185x>;
+		vdd-supply = <&tps_v3p3>;
+		drive-supply = <&tps_vposneg>;
+		vcom-supply = <&tps_vcom>;
+		/* panel nominal VCOM magnitude (VCOM itself is negative) */
+		vcom-microvolt = <3120000>;
 		control-device-name = "eink-panel";
 	};
 };
-- 
Armbian

